From f87b65f54e2a01a0be5e723289419b06870a8faf Mon Sep 17 00:00:00 2001
From: Daniel Wang <wonderfly@google.com>
Date: Wed, 28 Aug 2019 13:24:20 -0700
Subject: [PATCH] Fix systemd units dependencies

The unit definitions from upstream don't work for COS very well.
---
 .../src/lib/systemd/system/google-instance-setup.service      | 4 ++--
 .../src/lib/systemd/system/google-network-daemon.service      | 3 ++-
 .../src/lib/systemd/system/google-shutdown-scripts.service    | 3 ++-
 .../src/lib/systemd/system/google-startup-scripts.service     | 3 ++-
 4 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/packages/google-compute-engine/src/lib/systemd/system/google-instance-setup.service b/packages/google-compute-engine/src/lib/systemd/system/google-instance-setup.service
index ee987b2..5069b41 100644
--- a/packages/google-compute-engine/src/lib/systemd/system/google-instance-setup.service
+++ b/packages/google-compute-engine/src/lib/systemd/system/google-instance-setup.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Google Compute Engine Instance Setup
-After=network-online.target network.target rsyslog.service
-Before=sshd.service
+After=network-online.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/packages/google-compute-engine/src/lib/systemd/system/google-network-daemon.service b/packages/google-compute-engine/src/lib/systemd/system/google-network-daemon.service
index 71745d4..b4dd4bc 100644
--- a/packages/google-compute-engine/src/lib/systemd/system/google-network-daemon.service
+++ b/packages/google-compute-engine/src/lib/systemd/system/google-network-daemon.service
@@ -1,7 +1,8 @@
 [Unit]
 Description=Google Compute Engine Network Daemon
-After=network-online.target network.target
+After=network-online.target
 After=google-instance-setup.service
+Wants=network-online.target
 PartOf=network.service
 
 [Service]
diff --git a/packages/google-compute-engine/src/lib/systemd/system/google-shutdown-scripts.service b/packages/google-compute-engine/src/lib/systemd/system/google-shutdown-scripts.service
index ae23d76..c498de3 100644
--- a/packages/google-compute-engine/src/lib/systemd/system/google-shutdown-scripts.service
+++ b/packages/google-compute-engine/src/lib/systemd/system/google-shutdown-scripts.service
@@ -1,7 +1,8 @@
 [Unit]
 Description=Google Compute Engine Shutdown Scripts
-After=network-online.target network.target rsyslog.service
+After=network-online.target systemd-resolved.service
 After=google-instance-setup.service google-network-daemon.service
+Wants=network-online.target
 
 [Service]
 ExecStart=/bin/true
diff --git a/packages/google-compute-engine/src/lib/systemd/system/google-startup-scripts.service b/packages/google-compute-engine/src/lib/systemd/system/google-startup-scripts.service
index 233298a..68af491 100644
--- a/packages/google-compute-engine/src/lib/systemd/system/google-startup-scripts.service
+++ b/packages/google-compute-engine/src/lib/systemd/system/google-startup-scripts.service
@@ -1,7 +1,8 @@
 [Unit]
 Description=Google Compute Engine Startup Scripts
-After=network-online.target network.target rsyslog.service
+After=network-online.target
 After=google-instance-setup.service google-network-daemon.service
+Wants=network-online.target
 
 [Service]
 ExecStart=/usr/bin/google_metadata_script_runner --script-type startup
-- 
2.23.0.187.g17f5b7556c-goog

